projects/core/src/auth/user-auth/guards/custom-login.guard.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { CanActivate, GuardResult, Router } from '@angular/router';
+import { catchError, map, Observable, of, tap } from 'rxjs';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+} from '../../../global-message';
+import { SemanticPathService } from '../../../routing/configurable-routes/url-translation/semantic-path.service';
+import { StorageSyncType } from '../../../state/config/state-config';
+import {
+  getStorage,
+  persistToStorage,
+  readFromStorage,
+} from '../../../state/utils/browser-storage';
+import { WindowRef } from '../../../window/window-ref';
+import { AuthService } from '../facade/auth.service';
+import { CsrfStateService } from '../facade/csrf-state.service';
+
+const STORAGE_KEY = 'login_redirect_count';
+const timeout = 15_000;
+const totalRetries = 2;
```

Comment:
tested a few scenarios and see that 1 is enough (see as expected 2 failed requests and a redirect)
```suggestion
const totalRetries = 1;
```

Reply:
Updated, please verify.